docs/options.md

```diff
@@ -416,43 +416,26 @@ Default: reads your package's Python compatibility from `pyproject.toml`
 `setup.py` `setup(python_requires="...")`. If not found, cibuildwheel assumes
 the package is compatible with all versions of Python that it can build.
 
-
 !!! note
-    Rather than using this option, it's recommended you set
-    `project.requires-python` in `pyproject.toml` instead:
-
-    ```toml
-    [project]
-    requires-python = ">=3.6"
-    ```
+    Rather than using this option, it's recommended you set this value
+    statically in a way that your build backend can use it, too. This ensures
+    that your package's metadata is correct when published on PyPI.
 
-    A few things to note:
-
-      - If your project didn't already have a `pyproject.toml`, be sure to
-        set the build system to your build backend.
+      - If you have a `pyproject.toml` containing a `[project]` table, you can
+        specify `requires-python` there.
 
         ```toml
-        [build-system]
-        requires = ["setuptools>=42", "wheel"]
-        build-backend = "setuptools.build_meta"
+        [project]
+        requires-python = ">=3.6"
         ```
 
-      - If you didn't already have a `pyproject.toml` that had a `[project]`
-        table, you should migrate values from `setup.py` or `setup.cfg`, or
-        list them in the [the `dynamic` field](https://packaging.python.org/en/latest/specifications/declaring-project-metadata/#dynamic).
-        The values to include and the format is listed in the
-        [project source metadata specification](https://packaging.python.org/en/latest/specifications/declaring-project-metadata/#dynamic)
-        (as originally defined in PEP 621).
-
-      - Adding `[project]` to `pyproject.toml` can
-        change the behaviour of your build (e.g. `setuptools` may ignore
-        `install_requires` specified via `setup.py` or `setup.cfg`). Make sure
-        to double-check the build after adding.
-
-This option is not available in `pyproject.toml` under
-`tool.cibuildwheel.project-requires-python`, since it should be set with the
-[PEP621](https://www.python.org/dev/peps/pep-0621/) location instead,
-`project.requires-python`.
+        Note that build backend support for the `[project]` table is still patchy, and
+        adding `[project]` to `pyproject.toml` can change the behaviour of your build
+        (e.g. setuptools may ignore `install_requires` specified via `setup.py` or
+        `setup.cfg`). Make sure to double-check the build after adding.
```

Comment:
```suggestion
        Note that not all build backends fully support using a `[project]` table yet;
        specifically setuptools just added experimental support in version 61.
        Adding `[project]` to `pyproject.toml` requires all the other supported
        values to be specified there, or to be listed in `dynamic`.
```

Reply:
I'll take @henryiii's version, poetry isn't very relevant to wheel building, and it's the existence of the `[project]` table that invokes the change in behaviour.